lib/index.js

```diff
@@ -304,6 +304,14 @@ class PackWalker extends IgnoreWalker {
           file = file.slice(0, -2)
         }
         const inverse = `!${file}`
+
+        // if package.json's files array contains a ! and readme, we exclude non-root files
+        this.excludeNonRoot(file, inverse, 'readme', '!/readme{,.*[^~$]}')
+
+        // if package.json files array contains ! and either license/licence exclude non-root files
+        this.excludeNonRoot(file, inverse, 'license', '!/license{,.*[^~$]}')
+        this.excludeNonRoot(file, inverse, 'licence', '!/licence{,.*[^~$]}')
```

Comment:
what about `README`/`LICENSE` (different capitalizations)?

Reply:
excludeNonRoot will turn the file name to lowercase, so it works with different capitalizations.